File: flutter_sdk/test/ui/chat/view_models/chat_message_test.dart
// Copyright (c) Yalochat, Inc. All rights reserved.

import 'package:chat_flutter_sdk/src/ui/chat/view_models/chat_message.dart';
import 'package:test/test.dart';

void main() {
  group(ChatMessage, () {
    test('should return true when two equal chat messages are compared', () {
      final fixedTimestamp = DateTime(2025, 1, 1, 1);
      final chatMessage1 = ChatMessage(
        id: 0,
        role: MessageRole.user,
        type: MessageType.text,
        timestamp: fixedTimestamp,
      );
      final chatMessage2 = ChatMessage(
        id: 0,
        role: MessageRole.user,
        type: MessageType.text,
        timestamp: fixedTimestamp,
      );
      expect(chatMessage1.hashCode, equals(chatMessage2.hashCode));
      expect(chatMessage1, equals(chatMessage2));
    });

    test(
      'should return false when two different chat messages are compared',
      () {
        final fixedTimestamp = DateTime(2025, 1, 1, 1);
        final chatMessage1 = ChatMessage(
          id: 0,
          role: MessageRole.user,
          type: MessageType.text,
          timestamp: fixedTimestamp,
        );
        final chatMessage2 = ChatMessage(
          id: 0,
          role: MessageRole.system,
          type: MessageType.text,
          timestamp: fixedTimestamp,
        );
        expect(chatMessage1.hashCode, isNot(equals(chatMessage2.hashCode)));
        expect(chatMessage1, isNot(equals(chatMessage2)));
      },
    );

  });
}

File: android-sdk/sdk/src/commonTest/kotlin/com/yalo/chat/sdk/domain/model/ChatMessageTest.kt
// Copyright (c) Yalochat, Inc. All rights reserved.

package com.yalo.chat.sdk.domain.model

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertNotEquals

class ChatMessageTest {

    private val base = ChatMessage(
        role = MessageRole.USER,
        type = MessageType.Text,
        content = "hello",
        timestamp = 1000L,
    )

    @Test
    fun `two instances with identical fields are equal`() {
        val a = base.copy()
        val b = base.copy()
        assertEquals(a, b)
    }

    @Test
    fun `copy with single field change produces distinct instance`() {
        val modified = base.copy(content = "world")
        assertNotEquals(base, modified)
        assertEquals("world", modified.content)
        assertEquals(base.role, modified.role)
    }

    @Test
    fun `default values are applied when optional fields are omitted`() {
        val msg = ChatMessage(role = MessageRole.AGENT, type = MessageType.Text, timestamp = 0L)
        assertEquals("", msg.content)
        assertEquals(MessageStatus.IN_PROGRESS, msg.status)
        assertEquals(emptyList(), msg.products)
        assertEquals(emptyList(), msg.quickReplies)
        assertEquals(emptyList(), msg.amplitudes)
    }
}
